airgun/views/host.py

```diff
@@ -39,26 +40,68 @@ class TableActions(View):
 class PuppetClassParameterValue(Widget):
     """Represent value field for Puppet Class parameters table row from Host
     Parameters tab. That field can be interacted with as usual text input, but
-    also it can be overriden with new value
+    also it can be overridden with new value
     """
     ROOT = ".//div[@class='input-group']"
     value = TextInput(
-        locator=".//textarea[contains(@name, 'lookup_values_attributes')]")
-    override_parameter = Text(".//a[@data-tag='override']")
-    remove_override = Text(".//a[@data-tag='remove']")
+        locator=".//*[self::textarea or self::input][contains(@name, 'lookup_values_attributes')]")
+    override_button = Text(".//a[@data-tag='override']")
+    remove_override_button = Text(".//a[@data-tag='remove']")
+    hide_button = Text(".//a[contains(@class, 'btn-hide')]")
+
+    def _button_is_displayed(self, button_widget):
+        """Return whether the widget is displayed"""
+        return self.browser.wait_for_element(
+            button_widget, visible=True, exception=False, parent=self) is not None
+
+    @property
+    def hidden(self):
+        """Return whether the variable is hidden"""
+        return 'masked-input' in self.browser.classes(self.value)
+
+    @property
+    def overridden(self):
+        """Return whether the variable is overridden, a variable is overridden if not disabled"""
+        return not ('disabled' in self.browser.classes(self.value) or
```

Comment:
can you find single property to understand that one field is disabled?

Reply:
yes we can " 'disabled' in self.browser.classes(self.value) " can be sufficient for today usage